realm/realm-library/src/objectServer/java/io/realm/SyncConfiguration.java

```diff
@@ -1052,48 +1086,35 @@ public SyncConfiguration build() {
                 }
             }
 
-            // Check if the user has an identifier, if not, it cannot use /~/.
-            if (serverUrl.toString().contains("/~/") && user.getIdentity() == null) {
-                throw new IllegalStateException("The serverUrl contains a /~/, but the user does not have an identity." +
-                        " Most likely it hasn't been authenticated yet or has been created directly from an" +
-                        " access token. Use a path without /~/.");
-            }
-
             // Set the default Client Resync Mode based on the current type of Realm.
             // Eventually RECOVER_LOCAL_REALM should be the default for all types.
             if (clientResyncMode == null) {
-                clientResyncMode = ClientResyncMode.RECOVER_LOCAL_REALM;
+                clientResyncMode = ClientResyncMode.MANUAL;
             }
 
             if (rxFactory == null && isRxJavaAvailable()) {
                 rxFactory = new RealmObservableFactory(true);
             }
 
+            // FIXME: Figure out how to map to on-disk path. Partition key can be up to 16MB in size.
             // Determine location on disk
-            // Use the serverUrl + user to create a unique filepath unless it has been explicitly overridden.
-            // <rootDir>/<userIdentifier>/<serverPath>/<serverFileNameOrOverriddenFileName>
-            URI resolvedServerUrl = resolveServerUrl(serverUrl, user.getIdentity());
-            File rootDir = overrideDefaultFolder ? directory : defaultFolder;
-            String realmPathFromRootDir = user.getIdentity() + "/" + getServerPath(resolvedServerUrl);
-            File realmFileDirectory = new File(rootDir, realmPathFromRootDir);
-
-            String realmFileName = overrideDefaultLocalFileName ? fileName : defaultLocalFileName;
+            // Use the serverUrl + user to create a unique filepath.
+            // It is possible for users to change the root directory or realm file name.
```

Comment:
it is possible or not? if yes we need to add a test for it

Reply:
I removed the option for changing the realm file name. The Sync root dir is now being defined as part of the RealmAppconfiguration. There should be a test for it there.